Exiled.API/Features/Player.cs

```diff
@@ -1500,16 +1500,37 @@ public void Heal(float amount, bool overrideMaxHealth = false)
         /// <summary>
         /// Kills the player.
         /// </summary>
-        /// <param name="deathReason">The reason the player has been killed.</param>
-        /// <param name="cassieAnnouncement">The cassie announcement to make upon death.</param>
-        public void Kill(string deathReason, string cassieAnnouncement = "")
+        /// <param name="damageHandlerBase">The <see cref="DamageHandlerBase"/> used to kill.</param>
+        public void Kill(DamageHandlerBase damageHandlerBase)
         {
-            if (Side != Side.Scp && !string.IsNullOrEmpty(cassieAnnouncement))
-                Cassie.Message(cassieAnnouncement);
+            if (Side != Side.Scp && !string.IsNullOrEmpty(damageHandlerBase.CassieDeathAnnouncement.Announcement))
+                Cassie.Message(damageHandlerBase.CassieDeathAnnouncement.Announcement);
 
-            ReferenceHub.playerStats.KillPlayer(new CustomReasonDamageHandler(deathReason, float.MaxValue, cassieAnnouncement));
+            ReferenceHub.playerStats.KillPlayer(damageHandlerBase);
```

Comment:
We shouldn't use that method directly. We should use `DealDamage` with damage value set to `-1f`.
```cs
public void Kill(DamageHandlerBase damageHandlerBase)
{
    if (Side != Side.Scp && !string.IsNullOrEmpty(damageHandlerBase.CassieDeathAnnouncement.Announcement))
        Cassie.Message(damageHandlerBase.CassieDeathAnnouncement.Announcement);

    (damageHandlerBase as StandardDamageHandler).Damage = -1f;
    ReferenceHub.playerStats.DealDamage(damageHandlerBase);
}
```